doc/source/train_mnist.py

```diff
@@ -0,0 +1,419 @@
+# Copyright 2019 Mark van der Wilk, Vincent Dutordoir
```

Comment:
Do you expect this file to be included in the sphinx documentation? If not, would this file fit better in a dedicated `examples` directory or so?

Reply:
Yes, I was not expecting this to be in the Sphinx docs. Do you have any suggestions for a dir?

Reply:
GPflow.git/examples ?

Reply:
Though tbh I think a notebook would work better (it would make it more visible, could include some visualisations, and it would be included in the CI build to ensure it won't break unnoticedly at some point in the future).